src/primitives/transaction.cpp

```diff
@@ -111,5 +126,57 @@ std::string CTransaction::ToString() const
         str += "    " + tx_in.scriptWitness.ToString() + "\n";
     for (const auto& tx_out : vout)
         str += "    " + tx_out.ToString() + "\n";
+
+    if (!mweb_tx.IsNull()) {
+        str += "    " + mweb_tx.ToString() + "\n";
+    }
+
     return str;
 }
+
+std::vector<CTxInput> CTransaction::GetInputs() const noexcept
+{
+    std::vector<CTxInput> inputs;
+
+    for (const CTxIn& txin : vin) {
+        inputs.push_back(txin);
+    }
+
+    for (Commitment commit : mweb_tx.GetInputCommits()) {
```

Comment:
Is it possible to use Commitment& commit thereby avoiding a copy?

Reply:
Sure. I don't know if it will prevent a copy, since GetInputCommits builds a new set each time, but it might prevent a move. It depends on the compiler.